solana/pyth2wormhole/client/src/util.rs

```diff
@@ -0,0 +1,84 @@
+use log::trace;
+
+use tokio::sync::{Mutex, MutexGuard};
+use std::{ops::{Deref, DerefMut}, time::{Duration, Instant}};
+
+/// Rate-limited mutex. Ensures there's a period of minimum rl_interval between lock acquisitions
+pub struct RLMutex<T> {
```

Comment:
I'm really torn about this, because your snippet is significantly smaller, at the cost of some precision and user convenience I was really looking for. I think I may be falling for a sunken cost thing too, but here goes nothing :sweat_smile: 

This implementation may not be useful _right now_, but it may change when we decide to attest bigger symbol counts, which I think will happen long before the attester contract becomes obsolete. I'd argue the RLMutex take is more precise and harder to misuse, which may age better in that case.

Starting with the volume of the RLMutex, most is contributed by `RLMutexGuard` and `RLMutexState`, which primarily add one- or two-liner trait implementations.

Logic-wise, I don't think `rate_limit_lock` will achieve the desired behavior. Functionally, `RLMutex`  sleeps only if the desired duration hadn't passed _already_ since last lock release. If we do sleep, the wait time is precisely the remaining time until target rate limit is met, while `rate_limit_lock` will always slap a forced sleep onto each lock acquisition, no matter when it happens. Any attempt to match the desired behavior would result in keeping a mutable state equivalent to that present in `RLMutex`.

Finally, I would still prefer not to keep the rate value as a separate variable. I do not like that, because the job of applying the correct rate would move to each `rate_limit_lock` call site, and that makes each use prone to cause concurrency bugs, and IMO those tend to be a huge waste of time. If we were to keep the rate elsewhere, we would still end up with a wrapper for a `Mutex`, one way or another.